data/Transports.py

```diff
@@ -14,38 +50,124 @@ def temple_dest(boss) -> str:
+    "Tallon Overworld West (Root Cave)": RoomName.Transport_to_Magmoor_Caverns_East.value,
+    "Tallon Overworld East (Frigate Crash Site)": RoomName.Transport_to_Chozo_Ruins_East.value,
+    "Tallon Overworld South (Great Tree Hall, Upper)": RoomName.Transport_to_Chozo_Ruins_South.value,
+    "Tallon Overworld South (Great Tree Hall, Lower)": RoomName.Transport_to_Phazon_Mines_East.value,
+    "Chozo Ruins West (Main Plaza)": RoomName.Transport_to_Tallon_Overworld_North.value,
+    "Chozo Ruins North (Sun Tower)": RoomName.Transport_to_Magmoor_Caverns_North.value,
+    "Chozo Ruins East (Reflecting Pool, Save Station)": RoomName.Transport_to_Tallon_Overworld_East.value,
+    "Chozo Ruins South (Reflecting Pool, Far End)": "Chozo Ruins: " + RoomName.Transport_to_Tallon_Overworld_South.value,
+    "Magmoor Caverns North (Lava Lake)": RoomName.Transport_to_Chozo_Ruins_North.value,
+    "Magmoor Caverns West (Monitor Station)": RoomName.Transport_to_Phendrana_Drifts_North.value,
+    "Magmoor Caverns East (Twin Fires)": RoomName.Transport_to_Tallon_Overworld_West.value,
+    "Magmoor Caverns South (Magmoor Workstation, Save Station)": RoomName.Transport_to_Phendrana_Drifts_South.value,
+    "Magmoor Caverns South (Magmoor Workstation, Debris)": RoomName.Transport_to_Phazon_Mines_West.value,
+    "Phendrana Drifts North (Phendrana Shorelines)": RoomName.Transport_to_Magmoor_Caverns_West.value,
+    "Phendrana Drifts South (Quarantine Cave)": "Phendrana Drifts: " + RoomName.Transport_to_Magmoor_Caverns_South.value,
+    "Phazon Mines East (Main Quarry)": "Phazon Mines: " + RoomName.Transport_to_Tallon_Overworld_South.value,
+    "Phazon Mines West (Phazon Processing Center)": "Phazon Mines: " + RoomName.Transport_to_Magmoor_Caverns_South.value,
+}
 
-    }
 
-    return data[area]
+def get_transport_name_by_room_name(room_name: str) -> str:
+    for transport_name, room in _transport_names_to_room_names.items():
+        if room == room_name:
+            return transport_name
+    return room_name
+
+
+def get_room_name_by_transport_name(transport_name: str) -> str:
+    return _transport_names_to_room_names.get(transport_name, transport_name)
+
+
+def get_transport_data(world: 'MetroidPrimeWorld') -> Dict[str, Dict[str, str]]:
+    mapping = world.elevator_mapping
+    data = {}
+    for area in world.elevator_mapping.keys():
+        data[area] = {}
+        for source, dest in mapping[area].items():
+            data[area][get_transport_name_by_room_name(source)] = get_transport_name_by_room_name(dest)
+
+    data[MetroidPrimeArea.Tallon_Overworld.value]["Artifact Temple"] = temple_dest(world.options.final_bosses)
+    return data
+
+
+def get_region_by_elavator_name(elevator_name: str) -> str:
+    for region, elevators in default_elevator_mappings.items():
+        if elevator_name in elevators:
+            return region
+    return None
+
+
+def get_random_elevator_mapping(world: 'MetroidPrimeWorld') -> Dict[str, Dict[str, str]]:

```

Comment:
This is the function that will actually remap things (and account for denied connections as well)